src/arrayMethodJoin.js

```diff
@@ -5,7 +5,38 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let str = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (separator) {
+        if (this[i] === undefined || this[i] === null) {
+          this[i] = '';
+        }
+        str += `${this[i]}${separator}`;
+      }
+
+      if (separator === null) {
+        str += `${this[i] + 'null'}`;
+      }
+
+      if (separator === '') {
+        str += `${this[i]}`;
+      }
+
+      if (separator === undefined) {
+        str += `${this[i]},`;
+      }
+    }
+
+    if (this[this.length - 1]) {
+      str = str + this[this.length - 1];
+    }
```

Comment:
Looks too complicated, you need only two checks:
1. `if (this[i] !== null && this[i] !== undefined)` then we add this[i] to our str
2. `if (i !== this.length - 1)` : we check if curren index is not equal to the last index,  if it's not then we add separator to our str